fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/metric/ProtocolLogicalMetricGenerator.java

```diff
@@ -0,0 +1,201 @@
+     * <p>
+     * If the query contains undefined metrics, {@link BadApiRequestException} will be
+     * thrown.
+     *
+     * This method is meant for backwards compatibility. If you do not need to use this method for that reason please
+     * prefer using a generator instance instead.
+     *
+     * @param apiMetricQuery  URL query string containing the metrics separated by ','
+     * @param metricDictionary  Metric dictionary contains the map of valid metric names and logical metric objects
+     *
+     * @return set of metric objects
+     */
+    public LinkedHashSet<LogicalMetric> generateLogicalMetrics(
+            String apiMetricQuery,
+            MetricDictionary metricDictionary
+    ) {
+        LinkedHashSet<LogicalMetric> metrics = new LinkedHashSet<>();
+        List<String> invalidMetricNames = new ArrayList<>();
+
+        List<ApiMetric> apiMetrics = protocolAntlrApiMetricParser.apply(apiMetricQuery);
+        apiMetrics = apiMetrics.stream()
+                .map(apiMetricAnnotater)
+                .collect(Collectors.toList());
+
+        for (ApiMetric metric : apiMetrics) {
+            GeneratedMetricInfo generatedMetricInfo = new GeneratedMetricInfo(
+                    metric.getRawName(),
+                    metric.getBaseApiMetricId()
+            );
+
+            LogicalMetric baseLogicalMetrics = metricDictionary.get(metric.getBaseApiMetricId());
+            LogicalMetric logicalMetric = protocolChain.applyProtocols(generatedMetricInfo, metric, baseLogicalMetrics);
+            if (logicalMetric == null) {
+                invalidMetricNames.add(metric.getRawName());
+                continue;
+            }
+            logicalMetric = protocolChain.applyProtocols(generatedMetricInfo, metric, logicalMetric);
+            metrics.add(logicalMetric);
+        }
+        if (!invalidMetricNames.isEmpty()) {
+            String message = ErrorMessageFormat.METRICS_UNDEFINED.logFormat(invalidMetricNames);
+            LOG.error(message);
+            throw new BadApiRequestException(message);
+        }
+        return metrics;
+    }
+
+    /**
+     * Validate that all metrics are part of the logical table.
+     *
+     * This method is meant for backwards compatibility. If you do not need to use this method for that reason please
+     * prefer using a generator instance instead.
+     *
+     * @param logicalMetrics  The set of metrics being validated
+     * @param table  The logical table for the request
+     *
+     * @throws BadApiRequestException if the requested metrics are not in the logical table
+     */
+    public void validateMetrics(Set<LogicalMetric> logicalMetrics, LogicalTable table)
+            throws BadApiRequestException {
+        //get metric names from the logical table
+        Set<String> validMetricNames = table.getLogicalMetrics().stream()
+                .map(LogicalMetric::getLogicalMetricInfo)
+                .map(this::getBaseName)
```

Comment:
I think this is done in the wrong place. `validMetricNames` is pulling names off the logical table, none of which should be transformed.

I don't think using the base name is incorrect, but I don't think it is required.